community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/StatisticsBackedCardinalityModel.scala

```diff
@@ -42,6 +43,10 @@ class StatisticsBackedCardinalityModel(queryGraphCardinalityModel: QueryGraphCar
     case RegularQueryProjection(_, QueryShuffle(_, None, Some(limit: IntegerLiteral))) =>
       Cardinality.min(in, limit.value.toDouble)
 
+    // Normal projection with LIMIT
+    case RegularQueryProjection(_, QueryShuffle(_, None, Some(unknownLimit))) =>
+      Cardinality.min(in, GraphStatistics.DEFAULT_LIMIT_CARDINALITY)
+
```

Comment:
My thinking is that queries will change SKIP all the time, but will keep the same LIMIT. Why not auto-parameterize SKIP, but leave LIMIT as a literal? That seems the best balancing act to me. I expect most applications will have a limited range of LIMIT values, so the number of queries in the cache will be limited. Then we don't limit the limit planning at all!